src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4753,22 +4753,32 @@ WorkerLimitCount(Node *limitCount, Node *limitOffset, OrderByLimitReference
 	if (workerLimitNode != NULL && limitOffset != NULL)
 	{
 		Const *workerLimitConst = (Const *) workerLimitNode;
-		Const *workerOffsetConst = (Const *) limitOffset;
-		int64 workerLimitCount = DatumGetInt64(workerLimitConst->constvalue);
-		int64 workerOffsetCount = DatumGetInt64(workerOffsetConst->constvalue);
-
-		workerLimitCount = workerLimitCount + workerOffsetCount;
-		workerLimitNode = (Node *) MakeIntegerConstInt64(workerLimitCount);
+		if (!workerLimitConst->constisnull)
```

Comment:
Similar to LIMIT, OFFSET can also have NULL argument. `OFFSET NULL` is equivalent to `OFFSET 0` and it is safe to use DatumGetInt64 as we do it in the next block.

However, I think we need to explain this briefly in a comment.

Reply:
Added a comment.